conf/helmfile.d/0600.prometheus-operator.yaml

```diff
@@ -15,450 +15,121 @@ releases:
-
-      ## Provide custom recording or alerting rules to be deployed into the cluster.
-      ##
-      additionalPrometheusRules:
+    # A list of all possible values can be found:
+    # https://github.com/helm/charts/blob/master/stable/prometheus-operator/values.yaml
+    - additionalPrometheusRulesMap:
       - name: custom-prometheus-rules
         groups:
-          - name: custom-redis-metrics
+          - name: tf-serving-metrics
             rules:
-            - record: avg_gpu_usage
-              expr: avg(container_accelerator_duty_cycle)
+            - record: tf_serving_gpu_usage
+              expr: |-
+                avg(
+                  container_accelerator_duty_cycle{container_name="tf-serving"}
+                ) or vector(0)
+              labels:
+                deployment: tf-serving
+                namespace: deepcell
+            - record: tf_serving_up
+              expr: |-
+                max(
+                  clamp_max(
+                    kube_deployment_status_replicas_available{deployment="tf-serving"},
+                    1
+                  ) or vector(0)
+                )
               labels:
+                deployment: tf-serving
                 namespace: deepcell
-                service: gpu-average-service
-            - record: tf_serving_scale_metric
-              expr: avg_over_time(avg_gpu_usage[15s])
+          - name: consumer-metrics
+            rules:
+            - record: consumers_per_gpu
+              expr: |-
+                kube_deployment_status_replicas_available{deployment=~".*-consumer"}
+                / on() group_left
+                max(
+                  kube_deployment_status_replicas_available{deployment="tf-serving"}
+                  or vector(1)
+                )
               labels:
                 namespace: deepcell
-                service: tf-serving-scaling-service
-            - record: tf_serving_running_pods
+            - record: consumer_key_ratio
               expr: |-
-                clamp_max(
-                  clamp_min(
-                    kube_deployment_status_replicas_available{deployment="tf-serving"}
-                  , 0)
-                , 1)
-                or on()
-                up{service="kubernetes"} * on() 0
+                redis_script_values
+                / on(deployment)
+                kube_deployment_spec_replicas
               labels:
                 namespace: deepcell
-                service: tf-serving-running-service
             - record: segmentation_consumer_key_ratio
               # COMPLICATED scaling metric that prevents too many consumers
               # per GPU.  If too many consumers, scale down slightly.
               # Structural outline follows:
               # (
               #   min(1 - GPU, keys/consumers) * !is_too_many_consumers
               #   +
-              #   .9 * target * is_too_many_consumers)
+              #   .75 * target * is_too_many_consumers)
```

Comment:
Can we define numerical constants somewhere prominent in the file, similar to `conf/addons/hpa.yaml`?

Reply:
I believe I ran into issues doing this due to some issues doing that due to the helmfile templating, which works differently than gomplating.

Reply:
Maybe we could better template the rules by loading the rules files into a configmap (like the redis-exporter script) and reading the configmap inside this helmfile. Something to look into.

Reply:
I created #350 to track this, as I think its a good idea.

Reply:
Ok, I added a comment to #350 just to clarify that giving the numerical constants themselves reusable names would be really useful.